src/auth/jwtclient.ts

```diff
@@ -25,11 +25,11 @@ const isString = require('lodash.isstring');
 const noop = Function.prototype;
 
 export class JWT extends OAuth2Client {
-  public email: string;
-  public keyFile: string;
-  public key: string;
-  public scopes: string|string[];
-  public subject: string;
+  public email?: string;
+  public keyFile?: string|null;
```

Comment:
nit: likewise, it would be good if we could eliminate one of null vs. undefined if possible. Here and below.

Reply:
I hear you.  There are a few reasons I don't want to do this here.
1. For now - I just want this PR to represent the truth as we see it with the supported types.  We can come back around in follow up PRs and make it be what we want, but as of it now this is mostly documenting the types that are used.
2. These are both passed as constructor parameters, with tests that try passing null values.  So technically null is a valid value here.
3. There are places in the API where the JWT() is instantiated with no constructor parameters.  So technically undefined is valid as well.

I know this code is gross, but I'm scared to try touching too much in this already large PR.